java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -1369,6 +1369,9 @@ private void handlePackageProfileUpdate(MinionServer server,
                 .map(StateApplyResult::getChanges)
                 .filter(ret -> ret.getStdout() != null)
                 .map(CmdResult::getStdout);
+
+        ValueMap grains = new ValueMap(result.getGrains());
```

Comment:
IIUC moving the declaration here has the side effect that grains will always get updated for Ubuntu (see https://github.com/uyuni-project/uyuni/commit/939378ecb4570d89ebf698b389210281a3b89818)
Would that be a problem?

Reply:
I think this will not happen. The "if" statement for the update code block has not changed. We still test for `result.getGrains().isEmpty()` (see https://github.com/uyuni-project/uyuni/pull/3197/files#diff-eb3aa818a08d5c5af464fbc549102ad7ab37289784f95ee48750babe41091b59L1398)

Reply:
But I wonder if it would crash (even without my change). What will happen with this:

```
"ubuntu".equalsIgnoreCase((String) null))
```
If `result.getGrains()` is empty and we call `get("something")` on it, it will return `NULL`.

Reply:
Yes, I believe it would crash even without this PR.

Reply:
I will try to come up with a fix for this as well

Reply:
I use now the ValueMap to get the `os` grains. When it does not exist, it return an empty string.